src/coldfront_plugin_cloud/utils.py

```
@@ -149,21 +150,85 @@ def calculate_quota_unit_hours(allocation: Allocation,
+            after = get_included_duration(created, event_time, exclude_interval_list)
 
             value_times_seconds += (before * last_event_value) + (after * int(event.value))
             print(f"Last event at {last_event_time}, cr created at {created}, approved at {event_time}")
         else:
-            seconds_since_last_event = math.ceil((event_time - last_event_time).total_seconds())
+            seconds_since_last_event = get_included_duration(
+                last_event_time, event_time, exclude_interval_list)
             value_times_seconds += seconds_since_last_event * last_event_value
 
         last_event_time = event_time
         unbounded_last_event_time = event.modified
         last_event_value = int(event.value)
 
     # The value remains the same from the last event until the end.
-    since_last_event = math.ceil((end - last_event_time).total_seconds())
+    since_last_event = get_included_duration(last_event_time, end, exclude_interval_list)
     value_times_seconds += since_last_event * last_event_value
 
     return math.ceil(value_times_seconds / 3600)
+
+def load_excluded_intervals(excluded_interval_arglist):
+    excluded_intervals_list = list()
+    for interval in excluded_interval_arglist:
+        start, end = interval.strip().split(",")
+        start_dt, end_dt = [datetime.datetime.strptime(i, "%Y-%m-%d") for i in [start, end]]
+        assert end_dt > start_dt, "Interval end date must be after start date"
+        excluded_intervals_list.append(
+            [
+                datetime.datetime.strptime(start, "%Y-%m-%d"),
+                datetime.datetime.strptime(end, "%Y-%m-%d")
+            ] 
+        )
+
+    return excluded_intervals_list
+
+def _clamp_time(time, min_time, max_time):
+    if time < min_time:
+        time = min_time
+    if time > max_time:
+        time = max_time
+    return time
+
+def _merge_date_ranges(intervals_list: list):
```

I appreciate your enthusiasm in writing code to support this, but this then means that accepting overlapping intervals as an argument is a supported feature of the codebase, which we don't want to be the case.

Some explanations as to why:
- The admin running the command will not be incentivized to fix the way they are storing these intervals.
- It may also be the case that some other tools we're using do not implement these merging function, therefore using the same interval on OpenStack billing for example will return an error, but not here. 
- This is a complex function that we'd have to keep supporting moving forward in cases of bugs. 

Please simply check for overlapping intervals or unordered start and end and error in case of malformed input.

---

@knikolla Sorry for taking a while to ask this: Since removing support for merging intervals means overlapping intervals will result in faulty billing, I should throw an error for overlapping intervals as well?